src/controllers/web/GeographySearchController.ts

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+import Joi from 'joi';
+import { Lifecycle, Request, ResponseObject, ResponseToolkit } from '@hapi/hapi';
+
+import { geographyQuestionnaireOptions } from '../../data/geographyQuestionnaireOptions';
+import { transformTextInputError } from '../../utils/transformErrors';
+import { formIds, webRoutePaths } from '../../utils/constants';
+
+const GeographySearchController = {
+  renderGeographySearchHandler: async (request: Request, response: ResponseToolkit): Promise<ResponseObject> => {
+    const formFields = { ...geographyQuestionnaireOptions };
+    const {
+      geographySearch: geographySearchPath,
+      guidedDateSearch: guidedDateSearchPath,
+      results: resultsPath,
+      getResultsCount: resultsCountPath,
+    } = webRoutePaths;
+    const formId: string = formIds.geographyQuestionnaire;
+    return response.view('screens/guided_search/geography_questionnaire', {
+      guidedDateSearchPath,
+      geographySearchPath,
+      formFields,
+      formId,
+      resultsPath,
+      resultsCountPath,
+    });
+  },
+  doGeographySearchFailActionHandler: async (
+    request: Request,
+    response: ResponseToolkit,
+    error: Joi.ValidationError,
+  ): Promise<Lifecycle.ReturnValue> => {
+    const finalFormFields = await transformTextInputError({ ...geographyQuestionnaireOptions }, error);
+    const {
+      geographySearch: geographySearchPath,
+      guidedDateSearch: guidedDateSearchPath,
+      results: resultsPath,
+      getResultsCount: resultsCountPath,
+    } = webRoutePaths;
+    const formId: string = formIds.geographyQuestionnaire;
+    return response
+      .view('screens/guided_search/geography_questionnaire', {
+        guidedDateSearchPath,
+        geographySearchPath,
+        formFields: finalFormFields,
+        formId,
+        resultsPath,
+        resultsCountPath,
+      })
+      .code(400)
+      .takeover();
```

Comment:
I am unable to replicate this issue. However, as we discussed, the form resubmission still takes the old date if we refresh the page after validation with the latest form data. We can implement another approach, but this will give us inconsistent data due to distributed environments. Please suggest something on this.

Reply:
As discussed, the initial issue was unable to replicate. The other one we are ignoring for now.